### ISSUE TITLE
Is there a way to enable deno-mode only in deno projects?

### ISSUE THREAD

**Issue by finalclass** [rclarey/deno-emacs#3]
I'm working in various environments (node, browser, deno) and I cannot have deno-fmt enabled anytime I have a typescript-mode enabled. Currently I'm enabling it each time I open a new file and this is quite annoying. 
Is there a way to have it automatically enabled for all deno projects? (for example if in the file the is a reference to "deno.land" or if in the project we have "deno.jsonc" file?

**Comment by rclarey** [comment]
If you know that you'll always have a `deno.jsonc`/`deno.json` file for your deno projects that's probably the most reliable thing to check, and (assuming you use projectile) that can be done with something like:
```elisp
(defun deno-project-p ()
  (let ((root (projectile-project-root)))
    (unless (null root)
      (let ((config1 (concat root "deno.jsonc"))
            (config2 (concat root "deno.json")))
        (or (file-exists-p config1) (file-exists-p config2))))))

(defun fmt-for-deno ()
  (if (deno-project-p)
      (deno-fmt-mode)))

(add-hook 'typescript-mode-hook #'fmt-for-deno)
```

You could do a similar thing by searching for `deno.land` in the file, but that would only work for files with external imports.

**Comment by finalclass** [comment]
Thanks. You are awesome.
It would be wonderful if this would be a part of the deno-fmt plugin. Something like (global-deno-fmt-mode).

**Comment by rclarey** [comment]
To me that doesn't really fit the scope of this package. I think ideally there could be a package like Tide, but for Deno, that would do all the right setup for formatting, linting, LSP, etc. and it would have a check like above to only turn this stuff on for Deno projects.

I don't work with Deno that much currently, so I don't have the motivation to make such a package.

**Comment by finalclass** [comment]
IMO with TIDE there is a problem that it's not versatile enough. For example currently we have lsp-bridge but if we use TIDE we are stuck with it and cannot easily switch to lsp-bridge. Also making TIDE to work with deno is not that nice (one has to create package.json and essentially use npm). Moreover it's not easy to use deno-fmt with TIDE. Maybe that's possible but I lost some time on it and couldn't find a solution.

That's why I really like what you are doing in this package, namely: creating a collection of right configuration which anyone can use sparingly. To me it's a better approach.

Forgive my boldness but I'm currently slowly adopting deno and am curious what was your motivation to move away from it? Could you share it with me?

**Comment by rclarey** [comment]
>IMO with TIDE there is a problem that it's not versatile enough

Yeah that's a fair point

>Also making TIDE to work with deno is not that nice

I was just using TIDE as an example of a package that does many things, whereas this package is just focussed on letting you use `deno fmt` in Emacs. 

>Forgive my boldness but I'm currently slowly adopting deno and am curious what was your motivation to move away from it? Could you share it with me?

Oh, I'm not moving away from it. I use it for basically all of my personal projects, but I don't have much time to spend on those projects. I just comment/uncomment some of my config whenever I want to work on my Deno projects and that works well enough for me. If in the future I started working in Deno more then I would probably have the motivation to build a better config that switches between my Node and Deno configs automatically.



FYI I added the `deno-project-p` snippet above to the readme so it's more visible to others: https://github.com/rclarey/deno-emacs/tree/master#only-enable-deno-fmt-mode-for-deno-projects
